extensions/notification/common/client.mjs

```diff
@@ -28,7 +28,8 @@ class Client {
    * Note: This function returns immediately without waiting for the notification to disappear.
    */
   async s2c_showNotification(message, timeout) {
-    this.helper.mainUI.showNotification(message, timeout);
+    const san_message = filterXSS(message);
```

Comment:
Consider dealing with this directly in the mainUI.showNotification()/showAnnouncement() method, so that we don't have to change every user of it.

Reply:
Btw, it seems that we didn't check the type of `msg` to be String before we sanitize them.